paddle/ir/core/builtin_attribute_storage.h

```diff
@@ -114,43 +123,25 @@ struct ArrayAttributeStorage : public AttributeStorage {
   }
 
   bool operator==(const ParamKey &key) const {
-    if (key.size() != length_) {
-      return false;
-    }
-    for (size_t i = 0; i < length_; ++i) {
-      if (data_[i] != key[i]) {
-        return false;
-      }
-    }
-    return true;
+    return key.size() == size_ && std::equal(key.begin(), key.end(), data_);
   }
 
-  ParamKey GetAsKey() const { return ParamKey(data_, data_ + length_); }
-
- private:
-  Attribute *data_ = nullptr;
-  size_t length_ = 0;
-};
-
-struct TypeAttributeStorage : public AttributeStorage {
-  using ParamKey = Type;
-
-  explicit TypeAttributeStorage(const ParamKey &key) : value_(key) {}
-
-  static TypeAttributeStorage *Construct(ParamKey key) {
-    return new TypeAttributeStorage(key);
+  std::vector<Attribute> AsVector() const {
+    return std::vector<Attribute>(data_, data_ + size_);
   }
 
-  static std::size_t HashValue(const ParamKey &key) {
-    return std::hash<Type>()(key);
-  }
+  size_t size() const { return size_; }
 
-  bool operator==(const ParamKey &key) const { return value_ == key; }
+  bool empty() const { return size_ == 0u; }
 
-  ParamKey GetAsKey() const { return value_; }
+  Attribute at(size_t index) const {
+    IR_ENFORCE(index < size_, "Invalid index");
```

Comment:
```suggestion
    IR_ENFORCE(index < size_, "Index is out of range. Require index < size_, but received index.");
```
这里的报错信息可否完善下？（上述index和size_ 需要给出用户具体的值）

Reply:
好的。下个pr完善。